examples/provider/provider.tf

```diff
@@ -60,4 +60,10 @@ resource "hcp_consul_cluster" "example_secondary" {
   cluster_id   = "hcp-tf-example-consul-cluster-secondary"
   tier         = "development"
   primary_link = hcp_consul_cluster.example.self_link
+}
+
+// Create a Vault cluster in the same region and cloud provider as the HVN
+resource "hcp_vault_cluster" "example" {
+  cluster_id = "hcp-tf-example-vault-cluster"
+  hvn_id     = hcp_hvn.example_hvn.hvn_id
```

Comment:
Nit: since it's being added here, perhaps worth adding to the large example at the bottom of the main `README`?